src/replication/DealReplicationService.ts

```diff
@@ -0,0 +1,206 @@
+import ErrorCode from './ErrorCode';
+import GetReplicationDetailsResponse from './GetReplicationDetailsResponse';
+import { GetReplicationsResponse } from './GetReplicationsResponse';
+import UpdateReplicationRequest from './UpdateReplicationRequest';
+import { ObjectId } from 'mongodb';
+
+export default class DealReplicationService extends BaseService {
+
+    private app: Express = express();
+
+    public constructor () {
+      super(Category.DealReplicationService);
+      this.handleCreateReplicationRequest = this.handleCreateReplicationRequest.bind(this);
+      this.handleUpdateReplicationRequest = this.handleUpdateReplicationRequest.bind(this);
+      this.handleListReplicationRequests = this.handleListReplicationRequests.bind(this);
+      this.handleGetReplicationRequest = this.handleGetReplicationRequest.bind(this);
+      if (!this.enabled) {
+        this.logger.warn('Deal Replication Service is not enabled. Exit now...');
+        return;
+      }
+      this.app.use(Logger.getExpressLogger(Category.DealReplicationService));
+      this.app.use(bodyParser.urlencoded({ extended: false }));
+      this.app.use(bodyParser.json());
+      this.app.use(function (_req, res, next) {
+        res.setHeader('Content-Type', 'application/json');
+        next();
+      });
+      this.app.post('/replication', this.handleCreateReplicationRequest);
+      this.app.post('/replication/:id', this.handleUpdateReplicationRequest);
+      this.app.get('/replications', this.handleListReplicationRequests);
+      this.app.get('/replication/:id', this.handleGetReplicationRequest);
+    }
+
+    public start (): void {
+      const bind = config.get<string>('deal_replication_service.bind');
+      const port = config.get<number>('deal_replication_service.port');
+        this.app!.listen(port, bind, () => {
+          this.logger.info(`Deal Replication Service started listening at http://${bind}:${port}`);
+        });
+    }
+
+    private async handleGetReplicationRequest (request: Request, response: Response) {
+      const id = request.params['id'];
+      if (!ObjectId.isValid(id)) {
+        this.sendError(response, ErrorCode.INVALID_OBJECT_ID);
+        return;
+      }
+      this.logger.info(`Received request to get details of dataset replication request "${id}".`);
+      const found = await Datastore.ReplicationRequestModel.findById(id);
+      if (!found) {
+        this.sendError(response, ErrorCode.DATASET_NOT_FOUND);
+        return;
+      }
+      const result: GetReplicationDetailsResponse = {
+        id: found.id,
+        datasetId: found.datasetId,
+        replica: found.maxReplicas,
+        criteria: found.criteria,
+        client: found.client,
+        urlPrefix: found.urlPrefix,
+        maxPrice: found.maxPrice,
+        maxNumberOfDeals: found.maxNumberOfDeals,
+        isVerfied: String(found.isVerfied),
+        duration: found.duration,
+        isOffline: String(found.isOffline),
+        status: found.status
```

Comment:
Why not having those replicationTotal for this response?

Reply:
Create new issue to leave this for later https://github.com/tech-greedy/singularity/issues/70